client/components/edit-cards.jsx

```diff
@@ -18,6 +38,38 @@ class EditCards extends React.Component {
     });
   }
 
+  handleEditCard(event) {
+    event.preventDefault();
+    this.setState({
+      isUpdating: true
+    });
+    const { question, answer } = this.state;
+    const cardId = this.props.deck.cards[this.props.cardIndex].cardId;
+    if (question && answer) {
+      const reqBody = {};
+      reqBody.question = question;
+      reqBody.answer = answer;
+
+      const req = {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(reqBody)
+      };
+
+      fetch(`/api/card/${cardId}`, req)
+        .then(res => res.json())
+        .then(result => {
+          this.props.deck.cards[this.props.cardIndex].question = result.question;
+          this.props.deck.cards[this.props.cardIndex].answer = result.answer;
```

Comment:
In react apps, we don't assign / mutate our props like this. Instead pass the update info to a method that you receive from your parent as a prop, so that it can update the array immutably. This probably means passing the result object up to something like `this.props.onUpdated(result)`

Reply:
this also needs to be addressed